packages/flutter/lib/src/painting/notched_shapes.dart

```diff
@@ -0,0 +1,106 @@
+import 'dart:math' as math;
+
+import 'package:flutter/rendering.dart';
+
+/// A shape with a notch in its outline.
+///
+/// Typically used as the outline of a 'host' widget to make a notch that
+/// accomodates a 'guest' widget. e.g the [BottomAppBar] may have a notch to
+/// accomodate the [FloatingActionBar].
+abstract class NotchedShape {
```

Comment:
maybe have a see-also link to Shape in case anyone got lost when reading the docs

Reply:
done

Reply:
(I assume you meant ShapeBorder)